sdstore/sdstore.go

```diff
@@ -278,14 +299,15 @@ func (s *sdStore) putFile(url *url.URL, bodyType string, filePath string) error
 }
 
 // PUT request to SD store
-func (s *sdStore) put(url *url.URL, bodyType string, payload io.Reader, size int64) ([]byte, error) {
+func (s *sdStore) put(url *url.URL, bodyType string, payload io.Reader, size int64, md5s string) ([]byte, error) {
 	req, err := http.NewRequest("PUT", url.String(), payload)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Authorization", tokenHeader(s.token))
 	req.Header.Set("Content-Type", bodyType)
+	req.Header.Set("X-MD5", md5s)
```

Comment:
There is a size limit to header: https://stackoverflow.com/questions/24167656/nodejs-max-header-size-in-http-request
We might need to send this as a separate file that ties to each zip.